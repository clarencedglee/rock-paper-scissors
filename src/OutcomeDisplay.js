import React from 'react'

type Props = {
  outcome: -1 | 0 | 1 | undefined
};

export default (props: Props) => {
  // TODO: better visibility on no children? maybe throw?
  if (!props.children) return ''

  let content = typeof props.outcome === 'number'
    ? props.children[props.outcome + 2]
    : props.children[0]

  return <span className="outcome-display">{content}</span>
};