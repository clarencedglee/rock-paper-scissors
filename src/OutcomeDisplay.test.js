import React, {Fragment} from 'react';
import renderer from 'react-test-renderer';
import OutcomeDisplay from './OutcomeDisplay';

it('shows default message', () => {
    const component = renderer.create(
      <OutcomeDisplay outcome={undefined}>
        <Fragment>default</Fragment>
        <Fragment>lose</Fragment>
        <Fragment>tie</Fragment>
        <Fragment>win</Fragment>
      </OutcomeDisplay>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

it('shows lose message', () => {
    const component = renderer.create(
      <OutcomeDisplay outcome={-1}>
        <Fragment>default</Fragment>
        <Fragment>lose</Fragment>
        <Fragment>tie</Fragment>
        <Fragment>win</Fragment>
      </OutcomeDisplay>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

it('shows tie message', () => {
    const component = renderer.create(
      <OutcomeDisplay outcome={0}>
        <Fragment>default</Fragment>
        <Fragment>lose</Fragment>
        <Fragment>tie</Fragment>
        <Fragment>win</Fragment>
      </OutcomeDisplay>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

it('shows win message', () => {
    const component = renderer.create(
      <OutcomeDisplay outcome={0}>
        <Fragment>a</Fragment>
        <Fragment>b</Fragment>
        <Fragment>c</Fragment>
        <Fragment>d</Fragment>
      </OutcomeDisplay>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});