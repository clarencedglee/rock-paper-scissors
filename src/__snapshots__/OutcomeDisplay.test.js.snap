// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shows default message 1`] = `
<span
  className="outcome-display"
>
  default
</span>
`;

exports[`shows lose message 1`] = `
<span
  className="outcome-display"
>
  lose
</span>
`;

exports[`shows tie message 1`] = `
<span
  className="outcome-display"
>
  tie
</span>
`;

exports[`shows win message 1`] = `
<span
  className="outcome-display"
>
  c
</span>
`;
